### PR TITLE
cargo-dist: 0.5.0 -> 0.7.2

### PR DESCRIPTION
**Pull request by mistydemeo**
## Description of changes

This updates cargo-dist to the latest current release, 0.7.2.

I removed an extra set of package tests because these currently assume a git repo has been checked out. I'm a developer on the project and it's an oversight on our part; we've only tested `cargo test` from within a git checkout. I'll look into fixing it for a future version.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by lf- — APPROVED** [review]
lgtm. Please consider adding yourself to maintainers for the package given that you work on this upstream, so that you will receive pings if something goes awry or there are updates :)

In the next release, consider adding `passthru.updateScript = nix-update-script { };`, which will make the `r-ryantm` bot submit update PRs automatically and ping you with them. Currently there is a bug in nix-update which makes it catch prerelease versions, but I have a PR pending to fix that, and after that's merged, it's worth setting up auto-update on this package by adding that line. https://github.com/Mic92/nix-update/pull/221

Edit: for more info on the auto update infrastructure see this manual section: https://nixos.org/manual/nixpkgs/unstable/#var-passthru-updateScript

**Comment by mistydemeo** [comment]
> lgtm. Please consider adding yourself to maintainers for the package given that you work on this upstream, so that you will receive pings if something goes awry or there are updates :)

Sounds good - I'll do that in the next release!

> In the next release, consider adding passthru.updateScript = nix-update-script { };, which will make the r-ryantm bot submit update PRs automatically and ping you with them. Currently there is a bug in nix-update which makes it catch prerelease versions, but I have a PR pending to fix that, and after that's merged, it's worth setting up auto-update on this package by adding that line.

Interesting! Good to know, I'll do this in the next release once your fix is in.

**Review by matthiasbeyer — APPROVED** [review]

**Pull request merged**
